docs/resources/client/models/items.md

```diff
@@ -0,0 +1,1113 @@
+    itemModels.itemModelOutput.accept(
+        EXAMPLE_ITEM.get(),
+        new CompositeModel.Unbaked(
+            // The models to render
+            // Will render in the order they appear in the list
+            List.of(
+                new BlockModelWrapper.Unbaked(
+                    // Points to 'assets/examplemod/models/item/example_item_1.json'
+                    ResourceLocation.fromNamespaceAndPath("examplemod", "item/example_item_1"),
+                    // A list of tints to apply
+                    Collections.emptyList()
+                ),
+                new BlockModelWrapper.Unbaked(
+                    // Points to 'assets/examplemod/models/item/example_item_2.json'
+                    ResourceLocation.fromNamespaceAndPath("examplemod", "item/example_item_2"),
+                    // A list of tints to apply
+                    Collections.emptyList()
+                )
+            )
+        )
+    );
+}
+```
+
+</TabItem>
+</Tabs>
+
+## Property Models
+
+Some items change their state depending on the data stored in their stack (e.g., pulling a bow, breaking an elytra, a clock when in a given dimension, etc.). To allow models to change based on state, item models can specify a property to keep track of and select a model based on that condition. There are three different types of property models: range dispatch, select, and conditional. Each of these act as a expression for some float, switch case, and boolean respectively.
+
+### Range Dispatch Models
+
+Range dispatch models are the most similar to the item override some pre-1.21.4. Effectively, the type defines some `RangeSelectItemModelProperty` to get some float to switch the model on. Each entry then has some threshold value which the float must be greater than to render. The model chosen is the one with the closest threshold value that is not over the property value (e.g., if the property values is `4` with thresholds `3` and `5`, then the model associated with `3` will be rendered). The available `RangeSelectItemModelProperty`s to use can be found in `RangeSelectItemModelProperties`.
+
+<Tabs>
+<TabItem value="json" label="JSON" default>
+
+```json5
+// For some item 'examplemod:example_item'
+// JSON at 'assets/examplemod/items/example_item.json'
+{
+    "model": {
+        "type": "minecraft:range_dispatch",
+
+        // The `RangeSelectItemModelProperty` to use
+        "property": "minecraft:count",
+        // A scalar to multiply to the computed property value
+        // If count was 0.3 and scale was 0.2, then the threshold checked would be 0.3*0.2=0.06
+        "scale": 1,
+        "fallback": {
+            // The fallback model to use if no threshold matches
+            // Can be any unbaked model type
+            "type": "minecraft:model",
+            // Points to 'assets/examplemod/models/item/example_item.json'
+            "model": "examplemod:item/example_item"
+        },
+
+        // ~~ Properties defined by `Count` ~~
```

Comment:
Is this a weird pseudo-header? This looks very out of place. Same with "Entries with threshold information" below